PV260_Project/PresentationLayer/Program.cs

```diff
@@ -40,6 +48,13 @@ public class Program
                     services.AddTransient<IDataSetUi, DataSetUi>();
                     services.AddTransient<IDataLoader, CsvFileLoader>();
                     services.AddTransient<IDiffComputer, DiffComputer>();
+                    services.AddTransient<IEmailSender, GmailSender>();
+                    services.AddTransient<IMessageBuilder, HtmlMessageBuilder>();
+
+                    var section = config.GetSection(nameof(SmtpSettings));
```

Comment:
Just one little improvement which may work. Instead of registering the settings as singleton, you can use something like `services.AddOptions<SmtpSettings>().Bind(config.GetSection(nameof(SmtpSettings)));` to register it as `IOptions`. Then when you need to inject it somewhere, inject `IOptions<SmtpSettings> settings` instead of `SmtpSettings settings`

Reply:
If it is singleton and it's changed anywhere, it will also have these changes everytime after that.

Reply:
good point